### PR TITLE
Support customized tracing tags from headers

### PR DESCRIPTION
**Pull request by morvencao**
Context: https://github.com/istio/istio/issues/13018

Envoy v2 API supports [request_headers_for_tags](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#config-filter-network-http-connection-manager-v2-httpconnectionmanager-tracing), which can be used to create customized tags for the active span from request headers, this PR is meant to add this feature to Istio API.

/cc @douglas-reid @objectiser PTAK

### CODE REVIEW

**Comment by objectiser** [comment]
@morvencao I think @douglas-reid 's [comment on this recent API PR](https://github.com/istio/api/pull/907#issuecomment-486732030) would also apply here. It would be even more relevant for this property, as it will most likely be workload (and even endpoint) specific.

File: networking/v1alpha3/virtual_service.proto [hunk 1]
```diff
@@ -443,6 +443,8 @@ message Headers {
     map<string, string> add = 2;
     // Remove a the specified headers
     repeated string remove = 3;
+    // Customized tracing tags from the specified headers
```
Comment (costinm) [inline networking/v1alpha3/virtual_service.proto]: Example ? Use case ? Reference to any tracing standards ? (in the comments)

**Comment by douglas-reid** [comment]
@morvencao have you seen the [proposal](https://tinyurl.com/y3lnlf7n) on adding support for these types of tracing controls to the Sidecar API?

**Comment by morvencao** [comment]
Thanks @douglas-reid  really great proposal, I will close this.

**Pull request closed without merging**